alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportExecuteMonitorAction.kt

```diff
@@ -87,7 +95,10 @@ class TransportExecuteMonitorAction @Inject constructor(
                     }
                 })
             } else {
-                val monitor = execMonitorRequest.monitor as Monitor
+                val monitor = when (user == null || user?.name.isNullOrEmpty()) {
+                    true -> execMonitorRequest.monitor as Monitor
+                    false -> (execMonitorRequest.monitor as Monitor).copy(user = user)
```

Comment:
If user has permission to run mointor, then they can run other user's monitor?

Reply:
The flow is same, if User roles has permission to run monitor and read permission for the index being monitored, yes.

Reply:
If user enabled user backend role filter for alerting, then they should not see/run other user's monitor? Even user has read permission for the index being monitored and run monitor permission, user should not be able to see other user's monitor on Kibana, so they can't execute other user's monitor from Kibana. But if they know other user's monitor id, they can execute the monitor?
What problem will this PR address ? Per my understanding, this PR will fix such case: user in monitor has no run monitor permission, but the person who execute the monitor has run monitor permission, for such case, this PR will allow that person to run monitor successfully.